content/en/docs/components/pipelines/author-a-pipeline/components.md

```diff
@@ -164,7 +164,77 @@ To create a containerized component, you must:
 **When to use?** Containerized Python components should be used any time your component is implemented as Python code, but cannot be written as a standalone Python function or you wish to organize source code outside of the component Python function definition.
 
 ### 3. Custom container components
-<!-- @zichuan-scott-xu to provide this section -->
+
+Container-based components make it easier to use the pre-existed container to build a KFP component. 
+The user will control the container entrypoint and is able to define the component by specifying their container, commands, and args in a `ContainerSpec` object with similar authoring experiences as defining Python components. 
+
+To define a container component, you must:
+1) Write your component’s code as a python function that returns a `dsl.ContainerSpec` object to specify the container image and the commands to be run in the container and wrap the function into a `@container_component` decorator. The function should satisfy the following requirements:
+
+    *   It should not use any code declared outside of the function definition.
```

Comment:
nit:
This is not strictly true, as it uses `dsl.ContainerSpec` which is declared outside of the function definition. Also, the next bullet about `dsl.ContainerSpec` conveys this idea: "the function should do nothing other than..."

WDYT about removing the first bullet and turning the second bullet into a paragraph?

Definitely a nit, so feel free to proceed as-is if you prefer it that way